src/Util/JsonLD.php

```diff
@@ -191,12 +191,10 @@ public static function fetchElementArray($array, $element, $key = '@id')
 		$elements = [];
 
 		foreach ($array[$element] as $entry) {
-			if (!is_array($entry)) {
+			if (!is_array($entry) || (is_null($key) && is_array($entry))) {
 				$elements[] = $entry;
-			} elseif (isset($entry[$key])) {
+			} elseif (!is_null($key) && isset($entry[$key])) {
```

Comment:
Please swap the conditions if `$key` value must be checked first. Also why do you have to check that `$entry` isn't an array or it is discarded?

Reply:
When no `$key` is present, then all content is taken. When `$key` is present, then only entries with that key are taken - or stuff that isn't an array.

See the test `testFetchElementArrayFoundID`. Depending on the implementers the data could be "flat" or inside some array.